src/pygeosx/unitTests/CMakeLists.txt

```diff
@@ -0,0 +1,53 @@
+#
+# Specify list of tests
+#
+
+
+# the following two lists must have equal length:
+# they specify the couples (python script, xml file) corresponding to each test
+set( pythonFiles
+	testPygeosxAcoustic.py
+   testWaveEquationAdjoint.py
+   )
+
+set( xmlFiles
+        acous_5x5x5.xml
+        twoLayers2000x2000x2000.xml
+   )
+
+set( dependencyList pygeosx )
+
+if ( GEOSX_BUILD_SHARED_LIBS )
+  set (dependencyList ${dependencyList} geosx_core )
+else()
+  set (dependencyList ${dependencyList} ${geosx_core_libs} )
+endif()
+
+if( ENABLE_CUDA )
+  set( dependencyList ${dependencyList} cuda )
+endif()
+
+# This macro parse a test file and extract test_ function from it to pass it to ctest
+macro(parse_test_file filename)
+  set(filename ${filename})
+  file(READ "${filename}" contents)
+  string(REPLACE "\n" ";" rows ${contents})
+  foreach(line ${rows})
+    string(REGEX MATCHALL "^ *def *test_([A-Za-z_0-9])+ *" found ${line})
+    if (found)
+      string(REGEX REPLACE "^ *def *test_(([A-Za-z_0-9])+) *" "\\1" name "${found}")
+      configure_file(${CMAKE_CURRENT_SOURCE_DIR}/pytest_template.sh.in
+        ${CMAKE_CURRENT_BINARY_DIR}/${name}.sh
+        FILE_PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
+      blt_add_test(NAME ${name} COMMAND ${CMAKE_CURRENT_BINARY_DIR}/${name}.sh)
+    endif()
+  endforeach()
+endmacro(parse_test_file)
```

Comment:
IMO this is the key question since
- This does the test management/discovery by hand. But that's what `pytest` and `geosxats` already do.
- On top of this, we let `ctest` manage the python tests one by one and tells `pytest` to do it. And this does not prevent us from relying on the `pytest` dependency, since our tests explicitly use it (and it's nice).

Doing this, I'd be surprised that we benefit from an integrated test management of `pytest`.

My personnal long term goal would clearly to let `pytest` do all the discovery, reporting, ...
And then extract from `geosxats` the specific HPC parts (submissions (`python dask`?), comparisons, rebaselines). Or maybe make a `geosxats` module for `pytest` like many exist!
There is a question of tests in parallel.
That would mean less code on our side and we focus on more scientific code.

Reply:
In a nutshell, here are tracks you may want to explore.
- Do not parse `python` using cmake.
- Try to use the test discovery feature that `pytest` provides out of the box.
- If you think it makes sense and if the implementation makes sense, you can add a single `run python test` target in `ctest`.
- Be sure to properly split _integrated tests_ and _unit tests_. _Integrated tests_ should be developed using the `geosxats` fwk, like the others integrated tests.

@cssherman did I forget something?

Reply:
I'm going to work on adding a couple of simple unit tests using the method described by @TotoGaz that we can build on.  Once this is in place, we will try to get pygeosx working with the geosxats integrated testing system